cpp/models/d_abm/quad_well.h

```diff
@@ -0,0 +1,270 @@
+     */
+    double adoption_rate(const Agent& agent, const Status& new_status) const
+    {
+        double rate = 0;
+        // get the correct adoption rate
+        const size_t well = well_index(agent.position);
+        auto map_itr      = m_adoption_rates.find({well, agent.status, new_status});
+        if (map_itr != m_adoption_rates.end()) {
+            const auto& adoption_rate = map_itr->second;
+            // calculate the current rate, depending on order
+            if (adoption_rate.influences.size() == 0) { // first order adoption
+                // contact independant rate
+                rate = adoption_rate.factor;
+            }
+            else { // second order adoption
+                // accumulate rate per contact with a status in influences
+                size_t num_contacts   = 0;
+                ScalarType influences = 0;
+                for (auto& contact : populations) {
+                    // check if contact is indeed a contact
+                    if (is_contact(agent, contact)) {
+                        num_contacts++;
+                        for (size_t i = 0; i < adoption_rate.influences.size(); i++) {
+                            if (contact.status == adoption_rate.influences[i].status) {
+                                influences += adoption_rate.influences[i].factor;
+                            }
+                        }
+                    }
+                }
+                // rate = factor * "concentration of contacts with status new_status"
+                if (num_contacts > 0) {
+                    rate = adoption_rate.factor * (influences / num_contacts);
+                }
+            }
+        }
+        // else: no adoption from agent.status to new_status exist
+        return rate;
+    }
+
+    /**
+     * @brief Perform one integration step of the diffusion process for a given Agent using the Euler-Maruyama method.
+     * @param[in] dt Step size.
+     * @param[in] agent Agent to be moved.
+     */
+    void move(const double /*t*/, const double dt, Agent& agent)
+    {
+        const auto old_well = well_index(agent.position);
+        if (std::find(m_non_moving_states.begin(), m_non_moving_states.end(), agent.status) ==
+                m_non_moving_states.end() &&
+            std::find(m_non_moving_regions.begin(), m_non_moving_regions.end(), old_well) ==
+                m_non_moving_regions.end()) {
+            Position p = {mio::DistributionAdapter<std::normal_distribution<double>>::get_instance()(m_rng, 0.0, 1.0),
+                          mio::DistributionAdapter<std::normal_distribution<double>>::get_instance()(m_rng, 0.0, 1.0)};
+
+            agent.position      = agent.position - dt * grad_U(agent.position) + (m_sigma * std::sqrt(dt)) * p;
+            const auto new_well = well_index(agent.position);
+            if (old_well != new_well) {
+                m_number_transitions[static_cast<size_t>(agent.status)](old_well, new_well)++;
```

Comment:
Can you add an agent (or change ones position) in the move test, so this line gets covered? Maybe check as well that number transitions = 0,...,0,1,0,...,0